dwave/system/composites/reversecomposite.py

```diff
@@ -0,0 +1,204 @@
+
+        """
+        child = self.child
+
+        if anneal_schedules is None:
+            return child.sample(bqm, **parameters)
+
+        vartype_values = list(bqm.vartype.value)
+        if 'initial_state' not in parameters:
+            initial_state = dict(zip(list(bqm.variables), np.random.choice(vartype_values, len(bqm))))
+        else:
+            initial_state = parameters.pop('initial_state')
+
+        if not isinstance(initial_state,dict):
+            raise TypeError("initial state provided must be a dict, received {}".format(initial_state))
+
+        if 'reinitialize_state' not in parameters:
+            parameters['reinitialize_state'] = True
+            parameters['answer_mode'] = 'histogram'
+
+        vectors = {}
+        for schedule_idx, anneal_schedule in enumerate(anneal_schedules):
+            sampleset = child.sample(bqm, anneal_schedule=anneal_schedule,initial_state = initial_state,
+                                     **parameters)
+
+            initial_state,_ = dimod.as_samples(initial_state)
+            vectors = _update_data_vector(vectors,sampleset,
+                                           {'initial_state':[initial_state[0]]*len(sampleset.record.energy),
+                                            'schedule_index':[schedule_idx]*len(sampleset.record.energy)})
+
+            if parameters['reinitialize_state']:
+                # if reinitialize is on, choose the lowest energy, most probable state for next iteration
+                gse = sampleset.first.energy
+                b = sampleset.record[sampleset.record.energy == gse]
+                b.sort(order='num_occurrences')
+                initial_state = dict(zip(sampleset.variables, b[-1].sample))
+            else:
+                # if not reinitialized, take the last state as the next initial state
+                initial_state = dict(zip(sampleset.variables, sampleset.record.sample[-1]))
+
+        samples = vectors.pop('sample')
+        return dimod.SampleSet.from_samples((samples, bqm.variables),
+                                bqm.vartype,
+                                info={'anneal_schedules': anneal_schedules},
+                                **vectors)
+
+
+class BatchReverseComposite(dimod.ComposedSampler):
+    """ Composite that accepts multiple samples to initialize from.
+
+    Args:
+       sampler (:obj:`dimod.Sampler`):
+            A dimod sampler
+
+    """
+
+    def __init__(self, child_sampler):
+        self._children = [child_sampler]
+
+    @property
+    def children(self):
+        return self._children
+
+    @property
+    def parameters(self):
+        param = self.child.parameters.copy()
+        param['initial_states'] = []
+        return param
+
+    @property
+    def properties(self):
+        return {'child_properties': self.child.properties.copy()}
+
+    def sample(self, bqm, **parameters):
+        """ Composite that accepts multiple initial states to reverse annealing from
```

Comment:
"""Sample the binary quadratic model using reverse annealing from multiple initial states."""